### PR TITLE
Add linux proc extras.

### PR DESCRIPTION
**Pull request by gphat**
# What's this PR do?
Adds new metrics from linux's `/proc`. The readme contains all the new info.

# Motivation
We like to watch these metrics!

# Notes

Both of these are running and reporting metrics as expected on a QA machine.

r? @franklinhu @rhwlo @remh

### CODE REVIEW

File: checks.d/linux_mem_extras.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            regexp = re.compile(r'^(\w+):\s+([0-9]+)')  # We run this several times so one-time compile now
+            meminfo = {}
+
+            for line in lines:
+                try:
+                    match = re.search(regexp, line)
+                    if match is not None:
+                        meminfo[match.group(1)] = match.group(2)
+                except Exception:
+                    self.logger.exception("Cannot parse /proc/meminfo")
```
Comment (rhwlo) [inline checks.d/linux_mem_extras.py]: Could we change this error message to something like this?
```
self.logger.exception("Parsing error on /proc/meminfo line: %s" % line)
```

File: checks.d/linux_mem_extras.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+            try:
+                self.gauge('linux.memory.slab', int(meminfo.get('Slab', 0)) / 1024, tags)
```
Comment (rhwlo) [inline checks.d/linux_mem_extras.py]: Could we change this (and the next few lines) to:

```
STATS_ABOUT_WHICH_WE_GIVE_A_SHIT = ('Slab', 'PageTables', 'SwapCached')
⋮
  for sawwgas in STATS_ABOUT_WHICH_WE_GIVE_A_SHIT:
    try:
      self.gauge('linux.memory.%s' % sawwgas.lower(), int(meminfo.get(sawwgas, 0)) / 1024, tags)
    except Exception:
      self.logger.exception("Cannot compute stat `linux.memory.%s' from value `%i'" % (sawwgas.lower(), int(meminfo.get(sawwgas, 0)))
```

**Comment by rhwlo** [comment]
Solid work. I like it. :+1:

File: checks.d/linux_mem_extras.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    def check(self, instance):
+        if Platform.is_linux():
+            tags = instance.get('tags', [])
+
+            try:
+                with open('/proc/meminfo', 'r') as mem_info:
+                    lines = mem_info.readlines()
+
+            except Exception:
+                self.logger.exception('Cannot get memory metrics from /proc/meminfo')
```
Comment (remh) [inline checks.d/linux_mem_extras.py]: watchout the logger is `self.log` not `self.logger`

Comment (gphat) [inline checks.d/linux_mem_extras.py]: :bomb: Thanks!

**Comment by gphat** [comment]
Ok, bunch of stuff rolled in there. If you'd look again @rhwlo and maybe @remh if you have time that'd be great. ;)

**Comment by rhwlo** [comment]
All of this looks good to me.

**Pull request merged**